static/js/protein/chart.ts

```diff
@@ -432,6 +460,280 @@ export function drawProteinInteractionChart(
+  nodeData = nodeData.slice(0, MAX_INTERACTIONS); // consider only top 10 interactions to avoid clutter
+
+  const centerDatum = nodeFromID(centerNodeID, 0);
+  nodeData.push(centerDatum);
+
+  const linkData: InteractionLink[] = nodeData.map((node) => {
+    return {
+      score: node.value,
+      source: centerNodeID,
+      target: node.id,
+    };
+  });
+
+  const height = 400 - MARGIN.top - MARGIN.bottom;
+  const width = 700 - MARGIN.left - MARGIN.right;
+
+  const svg = d3
+    .select(`#${elementID}`)
+    .append("svg")
+    .attr("width", width + MARGIN.left + MARGIN.right)
+    .attr("height", height + MARGIN.top + MARGIN.bottom)
+    .attr("viewBox", `${-width / 2} ${-height / 2} ${width} ${height}`)
+    .append("g")
+    .attr("transform", `translate(${MARGIN.left - 150}, ${MARGIN.top - 25})`)
+    .attr("style", "max-width: 100%; height: auto; height: intrinsic");
+
+  const nodeIDs = nodeData.map((node) => node.id);
+  const nodeDepths = nodeData.map((node) => node.depth);
+  const nodeColors = d3.scaleOrdinal(nodeDepths, NODE_STYLE.circles.fillColors);
+
+  // force display layout
+  const forceNode = d3.forceManyBody();
+  const forceLink = d3
+    .forceLink(linkData)
+    .id(({ index }) => `${nodeIDs[index]}`);
+  forceLink.distance(LINK_STYLE.length);
+
+  const simulation = d3
+    .forceSimulation(nodeData)
+    .force("link", forceLink)
+    .force("charge", forceNode)
+    .force("center", d3.forceCenter())
+    .on("tick", ticked);
+
+  const links = svg // links first so nodes appear over links
+    .append("g")
+    .attr("stroke", LINK_STYLE.stroke.color)
+    .attr("stroke-opacity", LINK_STYLE.stroke.opacity)
+    .selectAll("line")
+    .data(linkData)
+    .join("line")
+    .attr(
+      "stroke-width",
+      (link) => LINK_STYLE.stroke.scoreWidthMultiplier * link.score
+    )
+    .attr("stroke-linecap", LINK_STYLE.stroke.linecap)
+    .on("mouseover", lightenThis)
+    .on("mouseleave", darkenThis);
+
+  const nodes = svg // container for circles and labels
+    .append("g")
+    .selectAll("g")
+    .data(nodeData)
+    .enter()
+    .append("g")
+    .call(drag(simulation))
+    .on("mouseover", lightenThis)
+    .on("mouseleave", darkenThis);
+
+  const nodeCircles = nodes
+    .append("circle")
+    .attr("stroke", NODE_STYLE.circles.stroke.color)
+    .attr("stroke-opacity", NODE_STYLE.circles.stroke.opacity)
+    .attr("stroke-width", NODE_STYLE.circles.stroke.width)
+    .attr("r", NODE_STYLE.circles.radius)
+    .attr("fill", (node) => nodeColors(node.depth));
+
+  const nodeLabels = nodes
+    .append("text")
+    .attr("fill", NODE_STYLE.labels.font.color)
+    .attr("text-anchor", "middle")
+    .attr("dominant-baseline", "middle")
+    .style(
+      "font",
+      `${NODE_STYLE.labels.font.size} ${NODE_STYLE.labels.font.name}`
+    )
+    .text(({ name }) => `${name}`);
+
+  // floating name labels we can switch in
```

Comment:
what do floating name labels look like? are you trying to decide between that and the current labels? maybe we can make the decision and not have to check in both the code

Reply:
Good question! They look something like this:
![image](https://user-images.githubusercontent.com/27650355/174910795-a5001b34-9eb2-4c2f-8b12-e229d54b2af4.png)

I think when the graph gets more crowded it may be easier to use smaller circles and floating labels.  However it's easier to tell which label is associated with which circle with the inside-circle labels.  Perhaps we can make this an option instead of a comment and later on change between floating and inside based on how crowded the graph is?

Reply:
do you have an example of a crowded graph? I feel like having a lot of floating labels when a graph is crowded could also get confusing ..

Reply:
Sadly I don't have such an example yet (will soon when I finish implementing breadth 2).  I'll just take out the floating label comment and we can revisit this question when the graph gets complicated enough to think about this with proper examples.